### PR TITLE
Bug: fix indexes bug with grahpql fields, multi-document issue and proper

### PR DESCRIPTION
**Pull request by magestrio**
…index name

## Type

- [ ] Feature
- [X] Bug fix
- [ ] CI/CD
- [ ] Documentation
- [ ] Other

## Description

Brief description of the changes made.

### CODE REVIEW

File: packages/serverless/src/apollo/app/collection-index.ts [hunk 1]
```diff
@@ -100,28 +95,22 @@ const indexExist = authorizeWrapper(
...
   async (_root: unknown, args: TIndexCreateRequest, ctx) =>
-    withTransaction((session) =>
-      createIndex(
-        args.databaseName,
-        ctx.userName,
-        args.collectionName,
-        args.indexField,
-        session
-      )
+    createIndex(
```
Comment (SangTran-127) [inline packages/serverless/src/apollo/app/collection-index.ts:98]: why did you drop withTransaction HOF? Does it need to use session?

File: packages/serverless/src/apollo/app/collection-index.ts [hunk 2]
```diff
@@ -100,28 +95,22 @@ const indexExist = authorizeWrapper(
...
   async (_root: unknown, args: TIndexDetailRequest, ctx) =>
-    withTransaction((session) =>
-      dropIndex(
-        args.databaseName,
-        ctx.userName,
-        args.collectionName,
-        args.indexName,
-        session
-      )
+    dropIndex(
```
Comment (SangTran-127) [inline packages/serverless/src/apollo/app/collection-index.ts:109]: same as above

File: packages/serverless/src/domain/use-case/collection.ts [hunk 3]
```diff
@@ -173,22 +165,22 @@ async function dropIndex(
...
-      'system',
-      session
-    )
+    await hasCollectionPermission(databaseName, collectionName, actor, 'system')
   ) {
-    // TODO: Should we check if index fields exist for a collection
-    return ModelCollection.getInstance(databaseName, collectionName).dropIndex(
-      indexName,
-      { session }
+    const index = `${indexName}_1`
```
Comment (SangTran-127) [inline packages/serverless/src/domain/use-case/collection.ts:173]: Is it hardcoded, for example, single index in mongoDB can be _1 or _-1 base on sort asc or desc

File: packages/serverless/src/domain/use-case/collection.ts [hunk 4]
```diff
@@ -173,22 +165,22 @@ async function dropIndex(
...
+    if (await doesIndexExist(databaseName, actor, collectionName, index)) {
+      // TODO: Should we check if index fields exist for a collection
```
Comment (SangTran-127) [inline packages/serverless/src/domain/use-case/collection.ts:175]: If you already check. You should remove the `TODO:`

File: packages/serverless/src/apollo/app/collection-index.ts (hunk 1 above)
Comment (magestrio) [reply packages/serverless/src/apollo/app/collection-index.ts:98]: We cannot create index withing transaction

File: packages/serverless/src/domain/use-case/collection.ts (hunk 4 above)
Comment (magestrio) [reply packages/serverless/src/domain/use-case/collection.ts]: removed

**Review by SangTran-127 — APPROVED** [review]

File: packages/serverless/src/domain/use-case/collection.ts (hunk 3 above)
Comment (SangTran-127) [reply packages/serverless/src/domain/use-case/collection.ts]: I think you should leave a TODO: to handle these index thing later

**Review by SangTran-127 — APPROVED** [review]

**Pull request merged**